rossum_api/api_client.py

```diff
@@ -81,6 +82,50 @@ async def authenticate_if_needed(self: APIClient, *args, **kwargs):
     return authenticate_if_needed
 
 
+class Retry:
+    """Retry configuration.
+
+    Requests will be retries up to `total` times with exponential backoff.
+    The backoff is applied after the second attempt and its length is determined
+    by following equation `backoff_factor * (2 ** (nth_attempt - 1))`.
+    """
+
+    def __init__(self, total: int = 3, backoff_factor: float = 1):
+        self.total = total
+        self.backoff_factor = backoff_factor
+
+    def sleep_for_attempt(self, attempt_i: int) -> float:
+        if attempt_i == 0:
+            return 0
+        return self.backoff_factor * (2 ** (attempt_i))
+
+    @staticmethod
+    def decorator(method):
+        """Decorate a method to retry the call if a possibly transient error occurs.
+
+        We retry all the request errors and server errors in 5xx range.
+        See https://www.python-httpx.org/exceptions/#the-exception-hierarchy for more details on request errors.
+        """
+
+        @functools.wraps(method)
+        async def retry(self: APIClient, *args, **kwargs):
+            for attempt in range(self.retry.total):
```

Comment:
It would make more sense to have `range(1, self.retry.total+1)` as I usually count my first attempt and not zeroth one 😅